dulwich/porcelain.py

```diff
@@ -191,6 +191,87 @@ class RemoteExists(Error):
     """Raised when the remote already exists."""
 
 
+def _parse_timezone_format(tz_str):
+    """Parse given string and attempt to return a timezone offset.
+    Different formats are considered in the following order:
+        - Git internal format: <unix timestamp> <timezone offset>
+        - RFC 2822: e.g. Mon, 20 Nov 1995 19:12:08 -0500
+        - ISO 8601: e.g. 1995-11-20T19:12:08-0500
+            - uses regex if dateutil is not available
+    Will default to UTC if timezone cannot be determined.
+    Args:
+      tz_str: datetime string
+    Returns: Timezone offset as integer
+    """
+    import re
+
+    # Git internal format
+    internal_format_pattern = re.compile("^[0-9]+ [+-][0-9]{,4}$")
+    if re.match(internal_format_pattern, tz_str):
+        try:
+            tz_internal = parse_timezone(tz_str.split(" ")[1].encode(DEFAULT_ENCODING))
+            return tz_internal[0]
+        except ValueError:
+            pass
+
+    # RFC 2822
+    import email.utils
+    rfc_2822 = email.utils.parsedate_tz(tz_str)
+    if rfc_2822:
+        return rfc_2822[9]
+
+    # ISO 8601
```

Comment:
Not a big fan of supporting two codepaths here. At the very least, could you make sure that both paths are covered by the testsuite?

Reply:
All kinds of cases I found so far are supported without needing dateutil. I was anticipating it to maybe catch some edge cases, but I guess we can also drop it?

Reply:
Yeah, let's just do that. If we find that there are other cases where we do want to rely on dateutil because handling them with custom code here is too complicated, we can always still add it later.

Reply:
Alright, done in 322ad9bc910b238f0e3d60ef9841fb3914a22aaa.